components/Pagination.tsx

```diff
@@ -0,0 +1,130 @@
+import Link from 'next/link';
+export const ITEMS_PER_PAGE = 24;
+
+type PaginationProps = {
+    totalItems: number;
+    url: string;
+    currentPage: number;
+    hasNextPage: boolean;
+    hasPreviousPage: boolean;
+};
+
+export function Pagination({ totalItems, currentPage, hasPreviousPage, hasNextPage, url }: PaginationProps) {
+    const totalPages = Math.ceil(totalItems / ITEMS_PER_PAGE);
+    const previousPage = currentPage === 1 ? 1 : currentPage - 1;
+    const nextPage = currentPage + 1;
+
+    const getPageUrl = (page: number) => `${url}?page=${page}`;
+
+    const renderPageNumbers = () => {
+        const pages: (number | string)[] = [];
+        const lastThreePages = [totalPages - 2, totalPages - 1, totalPages];
+
+        // Determine the starting page for the first segment
+        let startPage: number;
+        if (currentPage <= 3) {
+            startPage = 1;
+        } else if (currentPage >= totalPages - 4) {
+            startPage = totalPages - 6;
+        } else {
+            startPage = currentPage - 1;
+        }
+
+        // Add first three pages based on the start page
+        for (let i = 0; i < 3; i++) {
+            if (startPage + i <= totalPages - 3) {
+                pages.push(startPage + i);
+            }
+        }
+
+        // Add ellipsis if there's a gap
+        // @ts-expect-error
+        if (pages[pages.length - 1] < totalPages - 3) {
+            pages.push('...');
+        }
+
+        // Add last three pages
+        lastThreePages.forEach((page) => {
+            if (page > 0 && !pages.includes(page)) {
+                pages.push(page);
+            }
+        });
+
+        return pages.map((page, index) => {
+            if (page === '...') {
+                return (
+                    <span
+                        key={`ellipsis-${index}`}
```

Comment:
For React `key={`ellipsis-${index}`}` is === to `key={index}` so then I prefer the later one